clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesJobStatus.java

```diff
@@ -64,6 +71,45 @@ public KubernetesJobStatus(V1Job job, String account) {
     details.put("signal", this.signal != null ? this.signal.toString() : "");
     details.put("message", this.message != null ? this.message : "");
     details.put("reason", this.reason != null ? this.reason : "");
+
+    // find the execution details of the first pod that has failed. The Job completion details will
```

Comment:
it's a bit of a nit, but is there a reason not to return all of the pods statuses. If they're drilling down to that level of detail in the UI, we might as well give them everything we can about why a pod failed.

Reply:
I went with the thought that if the job has failed, then we should simply find the first pod that has containers which exit with a non-zero code and use its details in the spinnaker context. We could look at all pod failures, but it would start looking like a big wall of text in the ui. Atleast with one pod failure error, it'll be a good starting point for users to start debugging what went wrong.

Reply:
That makes sense. We discussed this at the Kubernetes SIG meeting on 1-26-21 and feedback from the community was very positive. This is a good place to start and if additional use cases bring up a need to retrieve more specific k8s objects then we can use this as a model.